addon/addon/models/sprite-snapshot-node-builder.ts

```diff
@@ -307,16 +307,53 @@ export class SpriteSnapshotNodeBuilder {
 
     // Classify non-natural KeptSprites
     for (let insertedSpriteModifier of classifiedInsertedSpriteModifiers) {
+      let counterpartSpriteModifier: SpriteModifier | undefined;
+
+      // find a suitable RemovedSprite counterpart if any
+      let removedSpriteModifiers = [...classifiedRemovedSpriteModifiers].filter(
+        (removedSpriteModifier) =>
+          new SpriteIdentifier(
+            insertedSpriteModifier.id,
+            insertedSpriteModifier.role
+          ).equals(
+            new SpriteIdentifier(
+              removedSpriteModifier.id,
+              removedSpriteModifier.role
+            )
+          )
+      );
+
+      assert(
+        'Multiple matching removedSpriteModifiers found',
+        removedSpriteModifiers.length < 2
+      );
+
+      let removedSpriteModifier = removedSpriteModifiers[0] as SpriteModifier;
+      if (removedSpriteModifier) {
+        classifiedRemovedSpriteModifiers.delete(removedSpriteModifier);
+      }
+
       let intermediateSprite = intermediateSprites.get(
         new SpriteIdentifier(
           insertedSpriteModifier.id,
           insertedSpriteModifier.role
         ).toString()
       );
-      if (intermediateSprite) {
+
+      if (intermediateSprite || removedSpriteModifier) {
+        classifiedKeptSpriteModifiers.add(insertedSpriteModifier);
+        classifiedInsertedSpriteModifiers.delete(insertedSpriteModifier);
+
+        // a matching IntermediateSprite always wins from a RemovedSprite counterpart
+        counterpartSpriteModifier =
+          intermediateSprite?.modifier ?? removedSpriteModifier;
+      }
+
+      if (counterpartSpriteModifier) {
+        // Find a stable shared ancestor AnimationContext
```

Comment:
These 2 if blocks can be consolidated into 1, and `counterpartSpriteModifier` can become a variable scoped to the consolidated block:

```
if (intermediateSprite || removedSpriteModifier) {
        classifiedKeptSpriteModifiers.add(insertedSpriteModifier);
        classifiedInsertedSpriteModifiers.delete(insertedSpriteModifier);

        // a matching IntermediateSprite always wins from a RemovedSprite counterpart
        let counterpartSpriteModifier =
          intermediateSprite?.modifier ?? removedSpriteModifier;
```